Update to k8s 1.35.2 and multus-cni v4.3.0

mgfritch opened the pull request:
**What this PR does / why we need it**:

The following dependabot PRs have become stale:
- https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller/pull/283
- https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller/pull/288
- https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller/pull/289
- https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller/pull/290
- https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller/pull/291

**Which issue(s) this PR fixes** 

- Remove explicit pin to k8s 1.29.1
- Update to k8s 1.35.2
- Update to multus-cni v4.3.0

**Special notes for your reviewer** *(optional)*:

The explicit pin to k8s 1.29.1 was required while using k8s.io/client-go v1.5.2 (obsolete), but it confuses the dependabot automation and it's no longer needed when using k8s.io/client-go v0.35.2

**Validation**

`make build && make test`

maiqueb commented:
I haven't forgotten, but this is quite big, and I need to take some time to go over it. I will try to provide feedback by the end of next week.

Huge thanks for the contribution. I'll keep in touch.

mgfritch commented on pkg/controller/pod.go:
```diff
@@ -77,7 +77,7 @@ type PodNetworksController struct {
 	broadcaster             record.EventBroadcaster
 	recorder                record.EventRecorder
-	workqueue               workqueue.RateLimitingInterface
+	workqueue               workqueue.TypedRateLimitingInterface[*string]
```
```suggestion
	workqueue               workqueue.TypedRateLimitingInterface[types.NamespacedName]
```

It might make sense to use `types.NamespacedName` here, but that's likely a larger refactor better suited for a new PR.

mgfritch commented:
> I haven't forgotten, but this is quite big, and I need to take some time to go over it. I will try to provide feedback by the end of next week.
> 
Thanks!

> Huge thanks for the contribution. I'll keep in touch.

The failing `golangci-lint` unit test should be resolved now. Please let me know if there's anything else I can do to help.

SchSeba commented on go.mod:
```diff
@@ -1,96 +1,68 @@
+	github.com/opencontainers/runtime-spec v1.2.1
+	google.golang.org/grpc v1.73.0
+	gopkg.in/k8snetworkplumbingwg/multus-cni.v4 v4.2.3
+	k8s.io/api v0.34.1
```
can you please move the k8s ones to v0.34.2

mgfritch replied:
Sorry for the delay, I've bumped the k8s client to v0.34.4

mgfritch committed:
Update to k8s 1.34.4, go 1.24.11 and multus-cni v4.2.4

- Remove explicit pin to k8s 1.29.1
- Update to go 1.24.11
- Update to k8s 1.34.4
- Update to multus-cni v4.2.4

Signed-off-by: Michael Fritch <mfritch@suse.com>

mgfritch committed:
Fix golangci-lint staticcheck errors

SA1019: workqueue.RateLimitingInterface is deprecated: Use TypedRateLimitingInterface instead. (staticcheck)
SA1019: workqueue.NewRateLimitingQueueWithConfig is deprecated: Use NewTypedRateLimitingQueueWithConfig instead. (staticcheck)
SA1019: workqueue.DefaultControllerRateLimiter is deprecated: Use DefaultTypedControllerRateLimiter instead. (staticcheck)
SA1019: workqueue.RateLimitingQueueConfig is deprecated: Use TypedRateLimitingQueueConfig instead. (staticcheck)

Signed-off-by: Michael Fritch <mfritch@suse.com>

mgfritch commented:
Updated the PR to use the latest k8s 1.34.4 and multus-cni v4.2.4

@maiqueb @SchSeba  can you please take another look?

SchSeba reviewed (approved):
LGTM

bpickard22 commented:
@maiqueb can you ptal, if you dont have time I can bypass the rules and merge but I would rather not

SchSeba commented:
Hi @maiqueb can you have a look on this one I would like to merge it if possible

SchSeba commented on go.mod:
```diff
@@ -1,96 +1,69 @@
-	k8s.io/cri-api v0.29.1
-	k8s.io/klog/v2 v2.120.1
-	k8s.io/kubelet v0.29.1
+	github.com/containernetworking/cni v1.3.0
```
Hi @mgfritch can you please take a look on this https://github.com/k8snetworkplumbingwg/ovs-cni/blob/main/pkg/types/types.go#L50

I had some issues with marshal after bumping to v1.3.0
I am not super familiar with this project code but please check if we are not doing any marshal function that can be impacted from this

mgfritch replied:
Thanks! I looked into this and I think we are safe since this controller doesn't embed `types.NetConf` (or any other upstream CNI structs).

maiqueb commented:
@mgfritch / @SchSeba I've updated golang to 1.25, since 1.24 is out of support already. 

Could you rebase ?... and adjust to that ?... I'll review it afterwards.

maiqueb commented:
I think this PR doesn't make sense after bumping golang to 1.25. 

Would you still be interested in upgrading the k8s API and multus used versions ?

SchSeba commented:
Hi @mgfritch can you please rebase and bump to 1.25 or even 1.26 that is already out with kubernetes version 1.36

mgfritch committed:
Update to k8s 1.35.2 and multus-cni v4.3.0

- Remove explicit pin to k8s 1.29.1
- Update to k8s 1.35.2
- Update to multus-cni v4.3.0

Signed-off-by: Michael Fritch <mfritch@suse.com>

mgfritch committed:
Fix golangci-lint staticcheck errors

SA1019: workqueue.RateLimitingInterface is deprecated: Use TypedRateLimitingInterface instead. (staticcheck)
SA1019: workqueue.NewRateLimitingQueueWithConfig is deprecated: Use NewTypedRateLimitingQueueWithConfig instead. (staticcheck)
SA1019: workqueue.DefaultControllerRateLimiter is deprecated: Use DefaultTypedControllerRateLimiter instead. (staticcheck)
SA1019: workqueue.RateLimitingQueueConfig is deprecated: Use TypedRateLimitingQueueConfig instead. (staticcheck)
SA1019: fake.NewSimpleClientset is deprecated: NewClientset replaces this with support for field management, which significantly improves server side apply testing. NewClientset is only available when apply configurations are generated (e.g. via --with-applyconfig). (staticcheck)

Signed-off-by: Michael Fritch <mfritch@suse.com>

mgfritch commented:
> Hi @mgfritch can you please rebase and bump to 1.25 or even 1.26 that is already out with kubernetes version 1.36

I rebased to pull in the bump to Go 1.25 and bumped k8s to 1.35.2 to keep us aligned with what's used by multus-cni v4.3.0.

@SchSeba @maiqueb PTAL, thanks!

SchSeba commented on pkg/controller/pod_test.go:
```diff
@@ -727,13 +727,6 @@ func newDummyPodController(
 			fmt.Fprintf(os.Stderr, "caches failed to sync (podInformerFactory): %v", v)
 		}
 	}
-	netAttachDefInformerFactory.Start(stopChannel)
```
why we don't need this one anymore?

mgfritch replied:
The fake client doesn't simulate bookmark events and will cause the test suite to deadlock/timeout. Since the definitions are static during the test run, we really don't need to actively watch for changes to the `NetworkAttachmentDefinitions`
